cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/LocalManager.java

```diff
@@ -107,6 +111,7 @@ public LocalManager(CConfiguration configuration, PipelineFactory pipelineFactor
     pipeline.addLast(new DeployDatasetModulesStage(datasetFramework));
     pipeline.addLast(new CreateDatasetInstancesStage(datasetFramework));
     pipeline.addLast(new CreateStreamsStage(id, streamAdmin, exploreFacade, exploreEnabled));
+    pipeline.addLast(new CreateSchedulesStage(store, scheduler));
```

Comment:
what is the stage that deletes the programs that are not in the new app spec, but were in the old one? Before that point (or in the same stage) we also have to remove the schedules for those programs, right? And after that we can add the new programs and the schedules for those?